testlib/resources.go

```diff
@@ -115,8 +121,23 @@ func SetupTestResources() string {
 	return tempDir
 }
 
-func CleanupTestResources(tempPath string) {
-	if tempPath != "" {
-		os.RemoveAll(tempPath)
+func setupConfig(configDir string) {
+	var err error
+
+	err = os.Mkdir(configDir, 0700)
+	if err != nil {
+		panic(fmt.Sprintf("failed to create config directory %s: %s", configDir, err.Error()))
```

Comment:
Since this is a helper method, I propose returning an `error` and panicking in `SetupTestResources` instead (or even earlier in the call chain, if you keep returning `error`). This avoids scattering panics throughout the code, and generally allows reuse of methods in other contexts without forcing panic semantics.

Reply:
(It also makes it easier to use `errors.Wrap` instead of having to `fmt.Sprintf` everywhere :))